primitives/linux-port/column.cpp

```diff
@@ -1420,67 +1614,162 @@ void vectorizedFiltering(NewColRequestHeader* in, ColResultHeader* out, const T*
     }
   }
 
-  // main loop
-  // writeMask tells which values must get into the result. Includes values that matches filters. Can have
-  // NULLs. nonEmptyMask tells which vector coords are not EMPTY magics. nonNullMask tells which vector coords
-  // are not NULL magics.
-  for (uint16_t i = 0; i < iterNumber; ++i)
+  if (in->hasAuxCol)
```

Comment:
Mb you will find some inspiration from the goldbot example I shared previously on how to make this part better.

Reply:
Done. I have changed this to use `if constexpr` semantics here and removed the code duplication in the if-else branches.